AngularJS/src/org/angular2/codeInsight/attributes/Angular2AttributeDescriptor.java

```diff
@@ -327,11 +328,23 @@ public PsiElement getDeclaration() {
           .flatCollection(inout -> newArrayList(inout.first.getNavigableElement(), inout.second.getNavigableElement()))
           .toList();
       case REGULAR:
-        return StreamEx.of(directive.getInputs())
-          .filter(input -> myInfo.name.equals(input.getName())
-                           && isOneTimeBindingProperty(input))
-          .map(Angular2Element::getNavigableElement)
-          .toList();
+        final Collection<PsiElement> propertiesPsiElements =
+              StreamEx.of(directive.getInputs())
+                      .filter(property -> {
+                        final boolean isOneTimeBinding = isOneTimeBindingProperty(property);
+                        return myInfo.name.equals(property.getName()) && isOneTimeBinding;
+                      })
+                      .map(Angular2Element::getNavigableElement)
+                      .toList();
+
+        final Collection<PsiElement> ctorParametersPsiElements =
```

Comment:
You could use `StreamEx.concat()` and avoid second toList() call.

Reply:
Yes that's better, thanks! I used `inputsStream.append(attributesStream).toList()`.